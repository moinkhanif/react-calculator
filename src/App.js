import { Component } from 'react';
import propTypes from 'prop-types';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor({ title }) {
    super({ title });
    this.title = title;
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit
            {' '}
            <code>src/App.js</code>
            {' '}
            and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            {this.title}
          </a>
        </header>
      </div>
    );
  }
}

App.propTypes = {
  title: propTypes.string,
};

App.defaultProps = {
  title: 'Calculator',
};

export default App;
