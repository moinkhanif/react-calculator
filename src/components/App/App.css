.main-container {
  max-width: 500px;
  margin: 10vh auto auto;
}
