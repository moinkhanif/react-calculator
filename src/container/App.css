.main-container {
  max-width: 700px;
  margin: 10vh auto auto;
  font-family: Arial, Helvetica, sans-serif;
  color: aliceblue;
}
