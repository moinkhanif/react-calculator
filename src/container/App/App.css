.main-container {
  max-width: 700px;
  margin: auto auto;
  font-family: Arial, Helvetica, sans-serif;
  color: aliceblue;
}
