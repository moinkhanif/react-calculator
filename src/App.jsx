import './App.css';
import Display from './components/Display/display.component';
import ButtonPanel from './components/ButtonPanel/button-panel.component';

const App = () => (
  <div className="main-container">
    <Display result="0" />
    <ButtonPanel />
  </div>
);

export default App;
