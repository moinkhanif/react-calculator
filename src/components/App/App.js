import { Component } from 'react';
import propTypes from 'prop-types';
import './App.css';

class App extends Component {
  constructor({ title }) {
    super({ title });
    this.title = title;
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          {this.title}
        </header>
      </div>
    );
  }
}

App.propTypes = {
  title: propTypes.string,
};

App.defaultProps = {
  title: 'Calculator',
};

export default App;
