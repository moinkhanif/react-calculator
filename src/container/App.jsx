import './App.css';
import { Component } from 'react';
import Display from '../components/Display/display.component';
import ButtonPanel from '../components/ButtonPanel/button-panel.component';
import calculate from '../components/logic/calculate';

class App extends Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = buttonName => {
    const name = buttonName.target.textContent;
    this.setState(state => calculate(state, name));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="main-container">
        <Display result={total} next={next} operation={operation} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default App;
