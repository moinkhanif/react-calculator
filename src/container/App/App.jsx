import './App.css';
import { useState } from 'react';
import Heading from '../../components/Heading/heading.component';
import Display from '../../components/Display/display.component';
import ButtonPanel from '../../components/ButtonPanel/button-panel.component';
import calculate from '../../components/logic/calculate';

const App = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const { total, next, operation } = state;

  const handleClick = buttonName => {
    const name = buttonName.target.textContent;
    setState(state => calculate(state, name));
  };

  return (
    <div className="main-container">
      <Heading />
      <Display result={total} next={next} operation={operation} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};

export default App;
