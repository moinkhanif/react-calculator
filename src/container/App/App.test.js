import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const result = renderer.create(<App />).toJSON();
  expect(result).toMatchSnapshot();
});
