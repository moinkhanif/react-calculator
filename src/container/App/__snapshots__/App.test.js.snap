// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="main-container"
>
  <div
    className="result"
  >
    0
  </div>
  <div
    className="groups"
  >
    <div
      className="group group1"
    >
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="Clear"
        type="button"
      >
        AC
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="plus/minus sign"
        type="button"
      >
        +/-
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="percent"
        type="button"
      >
        %
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="divide"
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="group group2"
    >
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="multiply"
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="group group3"
    >
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="subtract"
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="group group4"
    >
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="add"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="group group5"
    >
      <button
        className="calculator-key true orange"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="point"
        type="button"
      >
        .
      </button>
      <button
        className="calculator-key false orange"
        onClick={[Function]}
        title="equals to"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
